api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -88,6 +88,10 @@ type ClientTrafficPolicySpec struct {
 	//
 	// +optional
 	Timeout *ClientTimeout `json:"timeout,omitempty"`
+	// Connection provides configuration of client connection settings.
```

Comment:
thoughts on calling this `circuitBreaker` ?

Reply:
Yea, `Connection` is a very vague name... I feel like `CircuitBreaker` implies that some sort of failure or unexpected degradation in service triggers the blocking behavior, which isn't really the case here. WDYT about `Resources` or `Limits`?

Reply:
Some examples of how this feature is named in other products:
- Nginx: `limit_conn` - https://nginx.org/en/docs/http/ngx_http_limit_conn_module.html
- Kemp: `client connection limiter` (this is actually more of a connection rate limiter than a total connection limiter) - https://support.kemptechnologies.com/hc/en-us/articles/115004547103-How-To-Limit-The-Number-Of-Connections-Per-Second-From-A-Specific-Host
- HAProxy: `maxconn` - https://www.haproxy.com/documentation/haproxy-configuration-manual/latest/#maxconn
- Traefik: `InFlightConn` - https://doc.traefik.io/traefik/v3.0/middlewares/tcp/inflightconn/
- F5: `Connection Limits` - https://techdocs.f5.com/kb/en-us/products/big-ip_ltm/manuals/product/ltm-implementations-13-1-0/25.html
- Gloo Edge: `ConnectionLimits`- https://docs.solo.io/gloo-edge/main/guides/security/limit-connection/

So, I feel that `Connection[Limits|Settings|Options]` is a pretty intuitive name for a container which can also include other settings from the timeout and circuit breaker domains:

```yaml
connectionSettings:
  maxParallelConnections: 10000 # Total *[CircuitBreaker;Symmetric]*
  maxNewConnectionsPerSecond: 100 # [NEW;Downstream] CPS rate limit
  maxConnectionBufferSize: 100kb # *[NEW;Symmetric]* per connection memory allocation
  maxConnectionIdleTime: 600s # *[Timeout; Symmetric]*
  http:
    maxRequestsPerConnection: 5000 # *[CircuitBreaker; Symmetric]* close connection after this amount of request
    maxConnectionDuration: 3600s # *[CircuitBreaker; Symmetric]* close connection after this duration
    maxParallelRequests: 1000 # *[CircuitBreaker; Upstream]*  ATM (no envoy downstream equivalent that I know of, but could maybe be implemented through a simplified concurrency controller for adaptive concurrency filter. https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/adaptive_concurrency_filter#concurrency-controllers). 
    maxPendingRequest: 500 # *[CircuitBreaker; Upstream]* only for upstream, don't think that it makes sense for envoy to queue downstream requests... 
```

Reply:
thanks for doing the research !
I'm on board with the top level `connection` field now :) 
my only follow up since `limit` is not a struct but a value, how do we expose the `delay` knob, which is useful to thwart ddos